include/ocpp/common/message_queue.hpp

```diff
@@ -966,77 +1004,59 @@ template <typename M> class MessageQueue {
         }
     }
 
-    void set_registration_status_accepted() {
+    void set_registration_status_accepted() override {
         {
             std::lock_guard<std::recursive_mutex> lk(this->message_mutex);
             this->is_registration_status_accepted = true;
         }
         this->cv.notify_all();
     }
 
-    bool is_transaction_message_queue_empty() {
+    bool is_transaction_message_queue_empty() override {
         std::lock_guard<std::recursive_mutex> lk(this->message_mutex);
         return this->transaction_message_queue.empty();
     }
 
-    bool contains_transaction_messages(const CiString<36> transaction_id) {
-        std::lock_guard<std::recursive_mutex> lk(this->message_mutex);
-        for (const auto control_message : this->transaction_message_queue) {
-            if (control_message->messageType == v201::MessageType::TransactionEvent) {
-                v201::TransactionEventRequest req = control_message->message.at(CALL_PAYLOAD);
-                if (req.transactionInfo.transactionId == transaction_id) {
-                    return true;
-                }
-            }
-        }
```

Comment:
why is this removed?
The related change is
https://github.com/US-JOET/libocpp/commit/b72a6f601f62c8cdb80f6ffdd74f613ad12da4b4

which does not have any explanations

Reply:
It was not removed. It was moved into the v201 MessageQueue implementation (see lib/ocpp/v201/message_queue.cpp) since this logic is specific to v201::MessageType. Similarly with the v16::MessageType implementation which is in lib/ocpp/v16/message_queue.cpp.